src/Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -1,2180 +1,2682 @@
+
+                var fieldDefinitionHandle = metadataBuilder.AddFieldDefinition(
+                    enumTypeFieldAttributes,
+                    metadataBuilder.GetOrAddString("value__"),
+                    metadataBuilder.GetOrAddBlob(fieldSignature));
+                currentTypeDeclaration.AddField(symbol, fieldDefinitionHandle);
+
+                base.VisitEnumDeclaration(node);
+            }
+
+            ProcessTypeDeclaration(node, processEnumDeclaration);
+        }
+
+        public override void VisitEnumMemberDeclaration(EnumMemberDeclarationSyntax node)
+        {
+            base.VisitEnumMemberDeclaration(node);
+
+            var enumFieldAttributes =
+                FieldAttributes.Public |
+                FieldAttributes.Static |
+                FieldAttributes.Literal |
+                FieldAttributes.HasDefault;
+
+            var symbol = Model.GetDeclaredSymbol(node);
+            var fieldSignature = new BlobBuilder();
+            var encoder = new BlobEncoder(fieldSignature);
+            EncodeSymbol(new Symbol(symbol.ContainingType), encoder.FieldSignature());
+
+            var fieldDefinitionHandle = metadataBuilder.AddFieldDefinition(
+                enumFieldAttributes,
+                metadataBuilder.GetOrAddString(node.Identifier.Text),
+                metadataBuilder.GetOrAddBlob(fieldSignature));
+            currentTypeDeclaration.AddField(symbol, fieldDefinitionHandle);
+
+            if (symbol.HasConstantValue)
+            {
+                metadataBuilder.AddConstant(fieldDefinitionHandle, symbol.ConstantValue);
+            }
+        }
+
+        public override void VisitDelegateDeclaration(DelegateDeclarationSyntax node)
+        {
+            if (!IsPublicNode(node))
+            {
+                return;
+            }
+
+            var symbol = Model.GetDeclaredSymbol(node);
+            currentTypeDeclaration = new TypeDeclaration(symbol);
+
+            base.VisitDelegateDeclaration(node);
+
+            var objType = Model.Compilation.GetTypeByMetadataName(typeof(object).FullName);
+            var nativeIntType = Model.Compilation.GetTypeByMetadataName(typeof(IntPtr).FullName);
+
+            currentTypeDeclaration.AddMethod(
+                symbol,
+                ".ctor",
+                AddMethodDefinition(
+                    ".ctor",
+                    new Parameter[] {
+                        new Parameter(objType, "object", ParameterAttributes.None),
+                        new Parameter(nativeIntType, "method", ParameterAttributes.None)
+                    },
+                    null,
+                    false,
+                    false,
+                    true,
+                    false,
+                    true
+                )
+            );
+
+            int numParameters = node.ParameterList.Parameters.Count;
+            Parameter[] parameters = new Parameter[numParameters];
+            for (int idx = 0; idx < numParameters; idx++)

```

Comment:
want to pull this loop out into a function that we can call? seems like we do it a lot. just curious

Reply:
Good idea, addressed.